### PR TITLE
Support Node v18

### LINKED ISSUE #4061: Support Node v18
**Is your feature request related to a problem? Please describe.**

We should support the latest node version. It likely comes with a performance boost from an updated v8 + libuv.

**Describe the solution you'd like**

Add node 18 to our CI to ensure passing tests

### PR DESCRIPTION
**Pull request by dadepo**
**Motivation**

Support node version 18

**Description**

Fix any breaking changes and update CI to include v18 in build

Closes #4061

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -9,12 +9,16 @@ jobs:
   tests-main:
     name: Tests
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        node: ["lts/fermium", "lts/gallium", 18]
```
Comment (dapplion) [inline .github/workflows/test.yml]: @wemeetagain do we want to support NodeJS v14?

@dadepo Please add a comment like
```
lts/fermium = v14
lts/gallium = v16
```

Comment (philknows) [reply .github/workflows/test.yml]: I believe we had previous reports such as #4035 where not using Node v16 had issues. As of Nov 2021, active LTS was set at Node v16: https://github.com/ChainSafe/lodestar/pull/3440 and I don't believe there was any intent to keep supporting anything previous to v16?

Comment (wemeetagain) [reply .github/workflows/test.yml]: I think we drop support for 14 since its no longer supported upstream.

Also ideally we use numbers instead of release names for the node versions. fermium/gallium/etc is more confusing than it needs to be XD

Comment (dadepo) [reply .github/workflows/test.yml]: Yup. v14 is not supported. This is already an outdated diff, with version 14 removed in the commit [here](https://github.com/ChainSafe/lodestar/pull/4142/commits/c28e777b041e8942e94a137c90f76396d0de57ae)

@wemeetagain Indeed, version numbers are more straight-forward. I'll update existing usage of release names to version numbers.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -9,12 +9,16 @@ jobs:
   tests-main:
     name: Tests
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        node: [18]
```
Comment (dapplion) [inline .github/workflows/test.yml]: Why only support 18? We are using 16 everywhere

Comment (dadepo) [reply .github/workflows/test.yml]: Not only supporting 18. This is a temp commit as seen [here](https://github.com/ChainSafe/lodestar/pull/4142/commits/7e16c1f77a9d984b48220eaa1289a20728a11f64) while troubleshooting why the CI is failing. 

The PR is still in draft, so I'll suggest active PR review waits until when it is set to ready to review. 🙏

**Review by wemeetagain — DISMISSED** [review]

File: packages/api/test/utils/utils.ts [hunk 3]
```diff
@@ -14,15 +14,20 @@ export function getTestServer(): {baseUrl: string; server: FastifyInstance} {
...
       server.listen(port, function (err, address) {
-        if (err !== null && err != undefined) reject(err);
-        else resolve(address);
+        if (err !== null && err != undefined) {
+          reject(err);
+          console.log("error on listen", err);
+        } else {
+          console.log("listen on", address);
+          resolve(address);
+        }
```
Comment (wemeetagain) [inline packages/api/test/utils/utils.ts]: Is this change necessary?

Comment (dadepo) [reply packages/api/test/utils/utils.ts]: It was handy during debugging, and thought to leave it as it does not lead to a flood of log entries.

Comment (dadepo) [reply packages/api/test/utils/utils.ts]: Removed!

**Review by wemeetagain — APPROVED** [review]

**Comment by wemeetagain** [comment]
can you merge latest unstable on top?

File: packages/api/test/unit/beacon/debug.test.ts [hunk 4]
```diff
@@ -16,27 +16,29 @@ describe("beacon / debug", function () {
...
-    },
-    connectToPeer: {
-      args: ["peerId", ["multiaddr1", "multiaddr2"]],
-      res: undefined,
-    },
-    disconnectPeer: {
-      args: ["peerId"],
-      res: undefined,
-    },
+  describe("Run generic server test", () => {
```
Comment (dapplion) [inline packages/api/test/unit/beacon/debug.test.ts:19]: Why is the describe block necessary?

File: packages/lodestar/src/chain/blocks/importBlock.ts [hunk 5]
```diff
@@ -187,6 +192,35 @@ export async function importBlock(chain: ImportBlockModules, fullyVerifiedBlock:
     const cp = getCheckpointFromState(checkpointState);
     chain.checkpointStateCache.add(cp, checkpointState);
     pendingEvents.push(ChainEvent.checkpoint, cp, checkpointState);
+
+    // Note: in-lined code from previos handler of ChainEvent.checkpoint
+    chain.logger.verbose("Checkpoint processed", toCheckpointHex(cp));
```
Comment (dapplion) [inline packages/lodestar/src/chain/blocks/importBlock.ts]: Unwanted diff

**Review by dapplion — CHANGES_REQUESTED** [review]

File: packages/lodestar/src/chain/blocks/importBlock.ts (hunk 5 above)
Comment (dadepo) [reply packages/lodestar/src/chain/blocks/importBlock.ts]: This must be a result of the recent issues we have with our git merging strategies. If you look at the commits [here](https://github.com/ChainSafe/lodestar/pull/4142/commits) you see commits to this branch by you and @wemeetagain I don't think that is normal

@wemeetagain what do you suggest?

Comment (wemeetagain) [reply packages/lodestar/src/chain/blocks/importBlock.ts]: merge latest unstable on top

File: packages/api/test/unit/beacon/debug.test.ts (hunk 4 above)
Comment (dadepo) [reply packages/api/test/unit/beacon/debug.test.ts:19]: It made it possible to run just the generic server tests within the test file

**Review by dapplion — APPROVED** [review]

**Review by wemeetagain — APPROVED** [review]
